nncf/common/graph/patterns.py

```diff
@@ -260,3 +260,218 @@ def merge_two_types_of_operations(first_op: Dict, second_op: Dict, label: str) -
+    LINEAR_ELEMENTWISE = PatternDesc('linear_elementwise',
+                                 backends=[BackendType.OPENVINO],
+                                 devices=[TargetDevice.ANY, TargetDevice.CPU, TargetDevice.GPU])
+    LINEAR_BIASED_ELEMENTWISE = PatternDesc('linear_biased_elementwise',
+                                        backends=[BackendType.OPENVINO],
+                                        devices=[TargetDevice.ANY, TargetDevice.CPU, TargetDevice.GPU])
+    LINEAR_ACTIVATION_ELEMENTWISE = PatternDesc('linear_activation_elementwise',
+                                            backends=[BackendType.OPENVINO],
+                                            devices=[TargetDevice.ANY, TargetDevice.CPU, TargetDevice.GPU])
+    LINEAR_BIASED_ACTIVATION_ELEMENTWISE = PatternDesc('linear_biased_activation_elementwise',
+                                                   backends=[BackendType.OPENVINO],
+                                                   devices=[TargetDevice.ANY, TargetDevice.CPU, TargetDevice.GPU])
+
+    # OPERATIONS
+    LINEAR_OPERATIONS = PatternDesc('linear_operations', match=False)
+    BATCH_NORMALIZATION_OPERATIONS = PatternDesc('batch_normalization_operations', match=False)
+    ATOMIC_ACTIVATIONS_OPERATIONS = PatternDesc('atomic_activations_operations', match=False)
+    ARITHMETIC_OPERATIONS = PatternDesc('arithmetic_operations', match=False)
+
+    @staticmethod
+    def _get_pattern_desc_by_backend(backend: BackendType) -> List['PatternsManager']:
+        output = []
+        for pattern in PatternsManager:
+            pattern_desc = pattern.value
+            if pattern_desc.backends is None or backend in pattern_desc.backends:
+                output.append(pattern)
+        return output
+
+    @staticmethod
+    def _check(registered_patterns: List['PatternsManager'], all_patterns: List['PatternsManager']):
+        diff = set(all_patterns) - set(registered_patterns)
+        if len(diff) != 0:
+            raise RuntimeError('Not all patterns was registred in the backend!')
+
+    @staticmethod
+    def _filter_pattern_desc_by_device(pattern_names: List['PatternsManager'],
+                                       device: TargetDevice) -> List['PatternsManager']:
+        output = []
+        for backend_pattern in pattern_names:
+            pattern_desc = backend_pattern.value
+            if pattern_desc.devices is None or device in pattern_desc.devices:
+                output.append(backend_pattern)
+        return output
+
+    @staticmethod
+    def _get_fused_patterns_by_backend(backend: BackendType) -> Tuple[Registry, List['PatternsManager']]:
+        if backend == BackendType.OPENVINO:
+            from nncf.experimental.openvino_native.hardware.fused_patterns import \
+                OPENVINO_HW_FUSED_PATTERNS as ov_reg
+
+            pattern_names = PatternsManager._get_pattern_desc_by_backend(backend)
+            PatternsManager._check(ov_reg.registry_dict.keys(), pattern_names)
```

Comment:
Could you move this check to the test and exclude it from runtime just to save time?